### PR TITLE
61020 - Add logging to COE status endpoin

### PR DESCRIPTION
**Pull request by fred-khoury-oddball**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

We're still occasionally getting 4XX errors from LGY's /determination endpoint. I'm setting up logging to see if those errors are accompanied by any useful messages, and to doublecheck that we are indeed sending an EDIPI to LGY.

I'm on the benefits and claims team reviews is currently responsible for the COE form.

This work is behind a feature toggle currently enabled for 50% of vets.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#52015
- department-of-veterans-affairs/va.gov-team/issues/60243

## Testing done

- Updated specs

## Screenshots
_Note: Optional_

## What areas of the site does it impact?

COE form

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by DLarson-Oddball — COMMENTED** [review]
One question after an initial code inspection.

File: lib/lgy/service.rb [hunk 1]
```diff
@@ -51,6 +51,14 @@ def coe_status
         )
         nil
       end
+    rescue Common::Client::Errors::ClientError => e
+      log_message_to_sentry(
+        "GET coe_status failed with http status: #{e.status}",
+        :error,
+        { message: e.message, status: e.status, body: e.body, has_edipi: @edipi.present? },
+        { team: 'vfs-ebenefits' }
```
Comment (Midge-dev) [inline lib/lgy/service.rb:59]: Hey I created our Github team - https://github.com/orgs/department-of-veterans-affairs/teams/benefits-non-disability
Do we want to add it here?

File: spec/requests/v0/lgy_coe_request_spec.rb [hunk 2]
```diff
@@ -39,6 +39,23 @@
...
+          VCR.use_cassette 'lgy/determination_400_error' do
+            VCR.use_cassette 'lgy/application_not_found' do
+              expect_any_instance_of(LGY::Service).to receive(:log_message_to_sentry).with(
+                'GET coe_status failed with http status: 404',
+                :error,
+                { message: 'the server responded with status 404', status: 404, body: {}, has_edipi: true },
+                { team: 'vfs-ebenefits' }
+              )
+              get '/v0/coe/status'
+              expect(response.status).to eq(503)
```
[the remark this answers is not in the document]
Comment (fred-khoury-oddball) [reply spec/requests/v0/lgy_coe_request_spec.rb:54]: The 404 is the first expect statement that comes from the external LGY service. The 503 is an additional assertion saying after getting a response of 404 from the /determination endpoint we should expect a 503 from the vets-api endpoint /v0/coe/status. 
@DLarson-Oddball

File: lib/lgy/service.rb (hunk 1 above)
Comment (tblackwe) [reply lib/lgy/service.rb:59]: no, this is correct as vfs-ebenefits. it is describing the project in `sentry`

**Review by tblackwe — APPROVED** [review]

**Comment by tblackwe** [comment]
@fred-khoury-oddball whats the status of this PR?

**Pull request closed without merging**